Makefile

```diff
@@ -33,23 +39,46 @@ pre-pr: tidy generate lint test
 clean:
 	-@rm test/testdata/x.wasm
 
-sdk_srcs := $(wildcard ./basm/**/*.go)
-wasm_src_dir := $(shell realpath ./test/testdata)
-wasm_src := $(wasm_src_dir)/main.go
-wasm_out := $(wasm_src_dir)/x.wasm
+sdk_src := $(wildcard ./basm/**/*.go)
+wasm_src := $(testdata_dir)/main.go
+wasm_out := $(testdata_dir)/x.wasm
 
 .PHONY: wasm
 wasm: $(wasm_out)
 
-$(wasm_out): $(wasm_src) $(sdk_srcs)
+$(wasm_out): $(wasm_src) $(sdk_src)
 	@echo "Building WASM module..."
 	@tinygo build -o $@ -target=wasi $<
 
-test_scripts := $(shell realpath $(wildcard ./test/scripts/*.txtar))
+# Set default values for BASM_ variables. These can be overridden by the user
+# e.g. `BASM_USER_SECRET=mysecret make test`
+BASM_PLATFORM ?= plain
+BASM_CODE_MEASURE ?= plain
+BASM_AUTH_TOKEN ?= auth token
+BASM_HOST ?= local-server
+BASM_USER_SECRET ?= this is a bearer token
+# Export all BASM_ variables for use in recipes
+export $(filter BASM_%,$(.VARIABLES))
+
+template_src := $(wildcard $(template_dir)/*.tmpl)
+
+template_out := $(patsubst $(template_dir)/%.tmpl,$(rendered_template_dir)/%,$(template_src))
+# Mark the rendered files as intermediate. Make will delete them after the build.
+.INTERMEDIATE: $(template_out)
+
+# Rule to render template files, `envsubst <template_file> > <output_file>`
+$(rendered_template_dir)/%: $(template_dir)/%.tmpl
+	@envsubst < $< > $@
+
+.PHONY: test-integration
+test-integration: $(template_out) $(wasm_out)
+	@txtar-c $(rendered_template_dir) \
+		| cat "$(script_dir)/integration.txtar" - \
+		| testscript -e WASM_FILE=$(wasm_out)
```

Comment:
`test-integration` depends on `$(template_out)` which are generated files rendering config/input templates. These are generated in a temp dir and deleted after use. I organized it this way to hopefully help devs avoid checking in secrets as the rendered files may have secrets in them. 

The templated files are then concatenated into a txtar, concatenated with the test script, and passed to `testscript`